File: 프로그래머스/unrated/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.java
class Solution {
    public int solution(int[] arr) {
        int len = arr.length;
        int count = 0;
        int x = 0;
        while (count < len)
        {
            for (int i = 0; i < len ; i++)
            {
                if (arr[i] == 0)
                    continue;
                if (arr[i] < 50 && arr[i] % 2 == 1)
                    arr[i] = arr[i] * 2 + 1;
                else if (50 <= arr[i] && arr[i] % 2 == 0)
                    arr[i] /= 2;
                else {
                    count++;
                    arr[i] = 0;
                }
            }
            x++;
        }
        return x - 1;
    }
}

File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.js
function solution(arr) {
    var fct = ((v) => (v < 50 && v % 2 === 1) ? v * 2 + 1 : ((50 <= v && v % 2 === 0) ? v / 2 : v));
    var x = 0;
    var tmp = arr; 
    while (true) {
        var ttmp = tmp.map(v => fct(v));
        if (ttmp.every((v, i) => v === tmp[i]))
            break;
        tmp = ttmp;
        x++;
    }
    return x;
}